### PR TITLE
CommandModule: detect change based on update to some file

### PR DESCRIPTION
**Pull request by quinot**
When generating a file by running an ad hoc command on the target
host, allow detection of change based to a change in the contents
of a named file specified as a parameter to the action
(updates=<filename>).

### CODE REVIEW

File: commands/command.py [hunk 1]
```diff
@@ -188,12 +202,21 @@ def main():
...
         args = shlex.split(args)
     startd = datetime.datetime.now()
 
     rc, out, err = module.run_command(args, executable=executable, use_unsafe_shell=shell)
 
+    changed = True
+    if updates and updates_md5 is not None:
+        if updates_md5 == check_md5(module, updates_file, 'after', warnings):
+            changed = False
+
```
Comment (clonedagain) [inline commands/command.py]: How about a more direct writing? maybe :

```
    if updates and updates_md5 is not None:
        changed = (updates_md5 == check_md5(module, updates_file, 'after', warnings))
    else:
        # We didn't get a "updates" parameter, or we couldn't check the MD5 so by default we'll assume something changed
        changed = True
```

**Comment by clonedagain** [comment]
Apart from the style note I just wrote on the diff, :+1: I'd like to have this feature in Ansible and the patch looks reasonable (though I've not tested it yet).

**Comment by quinot** [comment]
clonedeagain: thanks for your comment, indeed your version is clearer (note: you need to change "==" to "!=", obviously).

**Comment by quinot** [comment]
Hello, do you have any target date for integration of this proposed enhancement?

**Comment by gregdek** [comment]
Hi @quinot. Thanks for the PR.

Because this module is a very central module, it's maintained by the Ansible Core team, and changes to these central modules can be conservative and very slow. Thanks for your patience with us here.

**Comment by gregdek** [comment]
@quinot Thanks for your continued patience here; the core team is trying to get 2.0 released, so we'll push this out until after 2.0.

**Comment by quinot** [comment]
@gregdek thanks for the update.

**Pull request closed without merging**